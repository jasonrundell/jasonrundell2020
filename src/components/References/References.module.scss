.list {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-bottom: $sizeNormal;
  padding: 0;
  list-style: none;
}

.item {
  margin-bottom: $sizeLarge;
}

.cite {
  color: $accentColorPrime;
}
