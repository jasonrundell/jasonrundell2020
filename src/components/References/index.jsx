import React from 'react'
import PropTypes from 'prop-types'
import { StaticQuery, graphql } from 'gatsby'

import Blockquote from '../Blockquote'
import Contentful from '../Contentful'

// CSS
import styles from './References.module.scss'

const { RichText } = Contentful

const References = ({ data }) => {
  const allReferences = data.allContentfulReferences.edges

  return (
    <>
      <ul className={styles.list}>
        {allReferences.map(reference => {
          const { id, citeName, company } = reference.node
          const quote = reference.node.quote.json

          return (
            <li key={id} className={styles.item}>
              <Blockquote>
                <RichText>{quote}</RichText>
              </Blockquote>
              <cite className={styles.cite}>
                - {citeName} ({company})
              </cite>
            </li>
          )
        })}
      </ul>
    </>
  )
}

export default props => (
  <StaticQuery
    query={graphql`
      {
        allContentfulReferences(sort: { fields: id }) {
          edges {
            node {
              id
              quote {
                json
              }
              citeName
              company
            }
          }
        }
      }
    `}
    render={data => <References data={data} {...props} />}
  />
)
References.propTypes = {
  data: PropTypes.shape({
    allContentfulReferences: PropTypes.shape({
      edges: PropTypes.arrayOf(PropTypes.object).isRequired,
    }).isRequired,
  }).isRequired,
}
