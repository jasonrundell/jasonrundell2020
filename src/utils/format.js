/**
 *  Letter case conversion functions
 *  [Letter case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
 *
 *  Converts only letter case-separated or hyphen-delimited strings
 */

/**
 *  Converts hyphen-delimited `string` to camel case
 *
 *  @example
 *  'foo-bar' => 'fooBar'
 *  'foo--bar' => 'fooBar'
 *
 *  @param {string} string
 */
function toCamelCase(string) {
  return string.replace(/(-+)([a-z])/g, ($0, $1, $2) => $2.toUpperCase())
}

/**
 *  Converts hyphen-delimited `string` to pascal case (aka upper camel case)
 *
 *  @example
 *  'foo-bar' => 'FooBar'
 *  'foo--bar' => 'FooBar'
 *
 *  @param {string} string
 */
function toPascalCase(string) {
  return toCamelCase(string.replace(/^[a-z]/, $0 => $0.toUpperCase()))
}

export { toCamelCase, toPascalCase }
