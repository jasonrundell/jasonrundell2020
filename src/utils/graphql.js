export function edgesToObject(edgesData) {
  const nodes = edgesData.map(item => {
    return item.node
  })
  const finalObject = nodes.map(item => {
    return item
  })

  return finalObject
}
