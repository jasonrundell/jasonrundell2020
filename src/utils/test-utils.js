export function catchWarnings() {
  const originalConsoleError = global.console.error
  beforeEach(() => {
    /**
     * I Want to know if values passed to props fail the type check
     * */
    global.console.error = (...args) => {
      const propTypeFailures = [/Warning:/] // e.g. [/Invalid prop/, /Failed prop type/]

      if (propTypeFailures.some(p => p.test(args[0]))) {
        throw new Error(args[0])
      }

      originalConsoleError(...args)
    }
  })
}

/**
 * Mix an match help from:
 * https://daveceddia.com/testing-for-proptypes-errors/
 * https://www.jackfranklin.co.uk/blog/failing-tests-on-react-proptypes/
 */
