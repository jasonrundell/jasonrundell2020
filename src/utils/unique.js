const onlyUnique = (value, index, self) => {
  return self.indexOf(value) === index
}

const generatePassword = ({
  passwordLength = 16,
  charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_-=~`',
}) => {
  let retVal = ''
  for (var i = 0, n = charset.length; i < passwordLength; ++i) {
    retVal += charset.charAt(Math.floor(Math.random() * n))
  }
  return retVal
}

export { onlyUnique, generatePassword }
