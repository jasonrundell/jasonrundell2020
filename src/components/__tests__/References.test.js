import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'
import { catchWarnings } from '../../utils/test-utils'

import References from '../References'

catchWarnings()

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

const testData = [
  {
    id: '79c48a1e-b7d2-54fe-8240-3b4544059f47',
    quote: {
      json: {
        data: {},
        content: [
          {
            data: {},
            content: [
              {
                data: {},
                marks: [],
                value:
                  'Jason brings a creativity to technology rarely seen in many developers. Jason continues to be a resource that I count on not only from a development perspective but also a trusted advisor for future development and the evolution of our technology platforms.',
                nodeType: 'text',
              },
            ],
            nodeType: 'paragraph',
          },
        ],
        nodeType: 'document',
      },
    },
    citeName: 'Gregg Tilston',
    company: 'Flight Centre Ltd.',
  },
]

describe('References component', () => {
  it('References renders correctly', () => {
    const tree = renderer.create(<References references={testData} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('References has required references prop', () => {
    act(() => {
      render(<References references={testData} />, container)
    })
  })
})
