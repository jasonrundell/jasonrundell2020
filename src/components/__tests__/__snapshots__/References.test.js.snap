// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`References component References renders correctly 1`] = `
<ul
  className="list"
>
  <li
    className="item"
  >
    <blockquote
      className="root"
    >
      <p
        className="root"
      >
        Jason brings a creativity to technology rarely seen in many developers. Jason continues to be a resource that I count on not only from a development perspective but also a trusted advisor for future development and the evolution of our technology platforms.
      </p>
    </blockquote>
    <cite
      className="cite"
    >
      - 
      Gregg Tilston
       (
      Flight Centre Ltd.
      )
    </cite>
  </li>
</ul>
`;
